spikeinterface/sortingcomponents/peak_detection.py

```diff
@@ -21,6 +21,116 @@
                    ('amplitude', 'float64'), ('segment_ind', 'int64')]
 
 
+
+def select_peaks(peaks, method='random', max_peaks_per_channel=1000, seed=None, **method_kwargs):
+
+    """Method to subsample all the found peaks before clustering
+
+    Parameters
+    ----------
+    peaks: the peaks that have been found
+    method: 'random', 'smart_sampling'
+        Method to use. Options:
+            * 'random' : a random subset is select from all the peaks
+            * 'smart_sampling' : peaks are selected by monte-carlo rejections
+    max_peaks_per_channel: int
+        The maximal number of peaks that should be kept per channel.
+    detect_threshold: float
+        Threshold, in median absolute deviations (MAD), to use to detect peaks.
+    seed: int
+        The seed for random generations
+    method_kwargs : dict
+        If the methods have particular parameters, they can be provided here
+    {}
+
+    Returns
+    -------
+    peaks: array
+        Detected peaks.
+
+    Notes
+    -----
+    This peak detection ported from tridesclous into spikeinterface.
+    """
+
+    selected_peaks = []
+    peaks_indices = {}
+
+    if seed is not None:
+        np.random.seed(seed)
+
+    for channel in np.unique(peaks['channel_ind']):
+        peaks_indices[channel] = np.where(peaks['channel_ind'] == channel)[0]
+
+    if method == 'random':
+        for channel in np.unique(peaks['channel_ind']):
+            selected_peaks += [np.random.permutation(peaks_indices[channel])[:max_peaks_per_channel]]
```

Comment:
You should use ```slected_inds = np.random.choice(peaks_indicies [channel].size, size=max_per_channel, replace=False)``` this avoid a full permutation and make a direct selection and so is faster.

Reply:
deal, done